### PR TITLE
Filtering events payload with beforeSend callback

### PR DESCRIPTION
**Pull request by khaydarov**

### CODE REVIEW

File: example.php [hunk 1]
```diff
@@ -14,7 +14,12 @@
 \Hawk\Catcher::init([
     'integrationToken' => 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwcm9qZWN0SWQiOiI2MDY2ZWI5N2U3NTU2ZDAwMjM2M2UyNjYiLCJpYXQiOjE2MTczNTc3MTl9.OpelHPPvS_TB8wUqCHRzcO3-Cp1VNL0UzlFuMfR35tk',
     'release'          => '12345',
-    'url'              => 'http://localhost:3000/'
+    'url'              => 'http://localhost:3000/',
+    'beforeSend'       => function (\Hawk\EventPayload $eventPayload) {
+        $eventPayload->setTitle('new title');
```
Comment (neSpecc) [inline example.php:19]: it would better to provide an example of how to remove fields

**Review by neSpecc — COMMENTED** [review]
and also, update readme please

File: README.md [hunk 2]
```diff
@@ -56,6 +56,21 @@ Use `sendEvent` method to send any data (logs, notices or something else)
 ]);
 ```
 
+### Filtering events data
```
Comment (neSpecc) [inline README.md:59]:
```suggestion
### Filtering sensitive information
```

File: README.md [hunk 3]
```diff
@@ -56,6 +56,21 @@ Use `sendEvent` method to send any data (logs, notices or something else)
...
+
+Add `beforeSend` callback to filter data before it is send to Hawk. Use setters to set any value or set empty to remove
```
Comment (neSpecc) [inline README.md:61]:
```suggestion
Use the `beforeSend` hook to filter any data you don't want to send to Hawk. Use setters to clear any property.
```

File: README.md [hunk 4]
```diff
@@ -56,6 +56,21 @@ Use `sendEvent` method to send any data (logs, notices or something else)
...
+
+```php
+\Hawk\Catcher::init([
+    // ...
+    'beforeSend' => function (\Hawk\EventPayload $eventPayload) {
+        $eventPayload->setTitle('');
```
Comment (neSpecc) [inline README.md:67]: Users will need a more realistic example. Something like that (didn't tested):

```php
$user = $eventPayload->getUser();

if (!empty($user['email'])){
  unset($user['email']);

  $eventPayload->setUser($user);
}   
```

File: example.php (hunk 1 above)
Comment (neSpecc) [reply example.php:19]: not responded

Comment (neSpecc) [reply example.php:19]: still actual question

Comment (khaydarov) [reply example.php]: done

**Review by neSpecc — APPROVED** [review]

**Review by talyguryn — APPROVED** [review]

**Pull request merged**